README.txt
Sobre

Este projeto tem como objetivo a avaliação de um sistema de comunicações sem fio quando especificidades das camadas
MAC/PHY são modeladas. Tal a avaliacao eh realiaza por meio de uma interface grafica.

#----------------------------------------------------------------------------------------------------------------------#

Objetivos
As metas desse tutorial são ajudar o usuário a:

    * Ter contato com as especificações de padrões de sistemas 3GPP para o 4G e o 5G;
    * Entender e protoipar especificidades das camadas MAC/PHY de sistemas modernos de comunicação móvel;
    * Avaliar a capacidade de pico de sistemas de comunicações móveis;
    * Comparar diferentes releases do 3GPP quanto a capacidade de pico.

#----------------------------------------------------------------------------------------------------------------------#

ESTRUTURA DO PROJETO

├── App
│   ├── Interface
│   │   ├── CalcTPutUI.py
│   │   ├── CalcTPutUI.ui
│   │   ├── __init__.py
│   ├── __init__.py
│   ├── app.py
│   ├── assets
│   │   └── tbs_size_table.csv
│   └── functions.py
├── CHANGELOG
├── README.txt
├── metadata.py
└── requirements.txt

#----------------------------------------------------------------------------------------------------------------------#

PREPARANDO O AMBIENTE:

    Dependencias:

        - Python 3.6.x ou maior

    Sistema Operacional:

        - Windows 10
        - Linux Ubuntu 20.04 LTS

    Windows:
    No prompt de comando.

        cd projeto02
        python -m venv venv
        venv\Scripts\activate.bat
        pip install -r requirements.txt

    Linux(Ubuntu):
    No terminal.

        cd projeto02
        sudo apt-get install python3-venv python3-tk libxkbcommon-x11-0 libx11-xcb1 libxcb-xinerama0
        python3 -m venv venv
        source venv/bin/activate
        pip install -r requirements.txt

#----------------------------------------------------------------------------------------------------------------------#

EXECUTANDO O PROJETO

    Dentro do diretorio raiz do projeto (ltecalculator), faca:

    Windows:
        - cd App
        - python app.py

    Linux:
        - cd App
        - python app.py

CHANGELOG
## [1.0.0] - 2020-07-23
### Adicionado
- Interface grafica
- Calculo do throughput pela tabela da norma
- Calculo do throughput pela equacao.
- metadata.py
- CHANGELOG
- Documentacao

## [1.0.1] - 2020-07-23
### Adicionado
- Informe de depencias do projeto da documentacao (README.txt)
- Comentarios para tornar o codigo mais legivel
- Formatacao de linhas.

## [1.0.2] - 2020-07-23
### Corrigido
- Retirando banda de 100MHz do combobox de banda

## [1.0.3] - 2020-07-24
### Adicionado
- Icone de janela principal
### Corrigido
- Tratamento de execao na classe principal.
- Documentacao

## [1.0.4] - 2020-07-24
### Removido
- import denecessario
### Corrigido
- Documentacao de execucao do programa no linux.

App/app.py
import sys
from pathlib import Path

from PyQt5.QtGui import QIcon
from PyQt5.QtWidgets import QMainWindow, QApplication

from Interface.CalcTPutUI import *
from functions import calc_lte_troughtput

data_folder = Path("assets/")
icon = data_folder / "lte-icon.png"


class CalcTPUT(QMainWindow, Ui_MainWindow):

    """
        Classe que instacia e faz utilizacao dos componentes da interface
    """

    def __init__(self, parent=None):

        super().__init__(parent)
        super().setupUi(self)
        self.setWindowTitle('Calculadora LTE-Advanced')
        self.setWindowIcon(QIcon("lte-icon.png"))

        # Pega o acionamento do bota Calcular da inteface e chama metodo calc_button_action()
        self.btnCalcTPut.clicked.connect(self.calc_button_action)

    def calc_button_action(self):

        """
        Metodo que colhe os valores dos campos na interface, chama a funcao de calculo do throughtput e retorna o valor
        na interface.
        :return:
        """

        bandwidth = self.comboBanda.currentText()
        mcs = self.comboMCS.currentText()
        mimo = self.ComboMimo.currentText()
        cp = self.comboPrefixCycle.currentText()
        ca = self.comboCarrierAgg.currentText()

        # Executa o calculo do throughput por meio da funcao calc_lte_troughtput
        calc_data = calc_lte_troughtput(bandwidth, mcs, mimo, cp, ca)

        try:

            self.outputPRB.setAlignment(QtCore.Qt.AlignCenter)
            self.outputPRB.setText(str(calc_data['PRBS']))

            self.outputTbsIndex.setAlignment(QtCore.Qt.AlignCenter)
            self.outputTbsIndex.setText(str(calc_data['TBSINDEX']))

            self.ouputTbsValue.setAlignment(QtCore.Qt.AlignCenter)
            self.ouputTbsValue.setText(str(calc_data['TBSVALUE']))

            self.outputModulacao.setAlignment(QtCore.Qt.AlignCenter)
            self.outputModulacao.setText(str(calc_data['MODULATION']) + 'QAM' if calc_data['MODULATION'] >= 16 else 'QPSK')

            self.outputNumRE.setAlignment(QtCore.Qt.AlignCenter)
            self.outputNumRE.setText(str(calc_data['NRE']))

            self.outputQtdSimbolos.setAlignment(QtCore.Qt.AlignCenter)
            self.outputQtdSimbolos.setText(str(calc_data['SYMBOLSQTD']))

            self.outputTabela.setText(f'{calc_data["TROUGHPUT_TABLE"]} Mbps')
            self.outputEquacao.setText(f'{calc_data["TROUGHPUT_EQUATION"]} Mbps')

        except TypeError as e:
            sys.exit(1)


if __name__ == '__main__':

    qt = QApplication(sys.argv)
    calctput = CalcTPUT()
    calctput.show()
    qt.exec_()


App/functions.py
import numpy as np
from pathlib import Path


# DICIONARIO CONTENDO A QUANTIDADE DE SIMBOLOS POR TIPO DO PREFIXO CILICO (NORMAL OU EXTENDIDO)
CYCLIC_PREFIX = {
    'Normal': {'SYMBOLS': 7},
    'Extendido': {'SYMBOLS': 6}
}

# DICIONARIO CONTENDO QUANTIDADE DE RESOURCE BLOCKS POR LARGURA DE BANDA UTILIZADA
PRB_PER_BW = {
    '1.4': (1.4, 6),
    '3': (3, 15),
    '5': (5, 25),
    '10': (10, 50),
    '15': (15, 75),
    '20': (20, 100),
    '100': (100, 100)

}

# DICIONARIO QUE MAPEIA A QUANTIDADE DE SIMBOLOS DA MODULACAO UTILIZADA, A TAXA DE CODIFICACAO E O INDICE DA TABELA TBS
# PELO VALOR DO MCS INDICAO NA INTERFACE
MODULATION_AND_CODE_RATE = {
    '0': {'MOD': 2, 'CR': 0.1172, 'TBSINDEX': 0}, '1': {'MOD': 2, 'CR': 0.1533, 'TBSINDEX': 1},
    '2': {'MOD': 2, 'CR': 0.1884, 'TBSINDEX': 2}, '3': {'MOD': 2, 'CR': 0.2451, 'TBSINDEX': 3},
    '4': {'MOD': 2, 'CR': 0.3007, 'TBSINDEX': 4}, '5': {'MOD': 2, 'CR': 0.3701, 'TBSINDEX': 5},
    '6': {'MOD': 2, 'CR': 0.4384, 'TBSINDEX': 6}, '7': {'MOD': 2, 'CR': 0.5136, 'TBSINDEX': 7},
    '8': {'MOD': 2, 'CR': 0.5879, 'TBSINDEX': 8}, '9': {'MOD': 2, 'CR': 0.6630, 'TBSINDEX': 9},
    '10': {'MOD': 4, 'CR': 0.3320, 'TBSINDEX': 9}, '11': {'MOD': 4, 'CR': 0.3691, 'TBSINDEX': 10},
    '12': {'MOD': 4, 'CR': 0.4238, 'TBSINDEX': 11}, '13': {'MOD': 4, 'CR': 0.4785, 'TBSINDEX': 12},
    '14': {'MOD': 4, 'CR': 0.5400, 'TBSINDEX': 13}, '15': {'MOD': 4, 'CR': 0.6015, 'TBSINDEX': 14},
    '16': {'MOD': 4, 'CR': 0.6425, 'TBSINDEX': 15}, '17': {'MOD': 6, 'CR': 0.4277, 'TBSINDEX': 15},
    '18': {'MOD': 6, 'CR': 0.4550, 'TBSINDEX': 16}, '19': {'MOD': 6, 'CR': 0.5049, 'TBSINDEX': 17},
    '20': {'MOD': 6, 'CR': 0.5537, 'TBSINDEX': 18}, '21': {'MOD': 6, 'CR': 0.6015, 'TBSINDEX': 19},
    '22': {'MOD': 6, 'CR': 0.6503, 'TBSINDEX': 20}, '23': {'MOD': 6, 'CR': 0.7021, 'TBSINDEX': 21},
    '24': {'MOD': 6, 'CR': 0.7540, 'TBSINDEX': 22}, '25': {'MOD': 6, 'CR': 0.8027, 'TBSINDEX': 23},
    '26': {'MOD': 6, 'CR': 0.8525, 'TBSINDEX': 24}, '27': {'MOD': 6, 'CR': 0.8886, 'TBSINDEX': 25},
    '28': {'MOD': 6, 'CR': 0.9257, 'TBSINDEX': 26}
}

# DICIONARIO QUE CONTEM O VALOR DO MIMO
MIMO = {'1': 1, '2': 2, '4': 4, '8': 8}


def calc_lte_troughtput(bandwidth, mcs, mimo_type, cyclic_prefix_type, component_carriers):

    """
    Funcao que realiza o calculo do throughput usando a tabela da norma e a equacao.

    :param bandwidth: Largura de banda
    :param mcs: MSC
    :param mimo_type: Tipo do mimo utilizado
    :param cyclic_prefix_type: Prefixo ciclico
    :param component_carriers: Carrier Agregation
    :return: dict()
    """

    try:
        # PEGA O CAMINHO DO ARQUIVO QUE CONTEM A TABELA DE MCSxPRBs
        data_folder = Path("assets/")
        csv_file = data_folder / "tbs_size_table.csv"

        # CONVERTE A TABELA ACIMA EM UM ARRAY PARA FACIL ACESSO.
        tbs_size_table = np.genfromtxt(csv_file, delimiter=",", skip_header=1, dtype=int)

        num_prbs = PRB_PER_BW[bandwidth][1]
        tbs_index = MODULATION_AND_CODE_RATE[mcs]['TBSINDEX']
        bits_from_tbs_size_table = tbs_size_table[tbs_index, int(num_prbs)]
        mimo = MIMO[mimo_type]
        cyclic_prefix = CYCLIC_PREFIX[cyclic_prefix_type]['SYMBOLS']
        ca = int(component_carriers)
        nre = 12 * cyclic_prefix

        # Calculo pela tabela
        troughput_table = (bits_from_tbs_size_table * ca * mimo * cyclic_prefix) / 7e3

        # Calculo pela equacao
        troughput_equation = (nre * MODULATION_AND_CODE_RATE[mcs]["MOD"] * mimo * 2 * num_prbs * 0.75 * ca) / 1_000

        return {
            'TROUGHPUT_TABLE': float(f'{troughput_table:.2f}'),
            'TROUGHPUT_EQUATION': float(f'{troughput_equation:.2f}'),
            'PRBS': num_prbs,
            'TBSINDEX': tbs_index,
            'TBSVALUE': bits_from_tbs_size_table,
            'MODULATION': 2 ** MODULATION_AND_CODE_RATE[mcs]["MOD"],
            'NRE': nre,
            'SYMBOLSQTD': cyclic_prefix
        }

    except OSError as e:
        print(f'Exception: {e}')
        print('Voce nao se encontra no diretorio correto.')
        print('Acesse ltecalculator/App e execute: python app.py')



App/Interface/CalcTPutUI.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'CalcTPutUI.ui'
#
# Created by: PyQt5 UI code generator 5.15.0
#
# WARNING: Any manual changes made to this file will be lost when pyuic5 is
# run again.  Do not edit this file unless you know what you are doing.


from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(593, 579)
        MainWindow.setStyleSheet("background: white;")
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.gridLayout = QtWidgets.QGridLayout(self.centralwidget)
        self.gridLayout.setObjectName("gridLayout")
        self.inputDataFrame = QtWidgets.QFrame(self.centralwidget)
        self.inputDataFrame.setStyleSheet("border: 2px solid #8f8f91; background: ")
        self.inputDataFrame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.inputDataFrame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.inputDataFrame.setObjectName("inputDataFrame")
        self.gridLayout_2 = QtWidgets.QGridLayout(self.inputDataFrame)
        self.gridLayout_2.setObjectName("gridLayout_2")
        self.label_5 = QtWidgets.QLabel(self.inputDataFrame)
        self.label_5.setStyleSheet("border: none;")
        self.label_5.setAlignment(QtCore.Qt.AlignCenter)
        self.label_5.setObjectName("label_5")
        self.gridLayout_2.addWidget(self.label_5, 3, 4, 1, 1)
        self.comboBanda = QtWidgets.QComboBox(self.inputDataFrame)
        self.comboBanda.setObjectName("comboBanda")
        self.comboBanda.addItem("")
        self.comboBanda.addItem("")
        self.comboBanda.addItem("")
        self.comboBanda.addItem("")
        self.comboBanda.addItem("")
        self.comboBanda.addItem("")
        self.gridLayout_2.addWidget(self.comboBanda, 4, 1, 1, 1)
        self.label_3 = QtWidgets.QLabel(self.inputDataFrame)
        self.label_3.setStyleSheet("border: none;")
        self.label_3.setAlignment(QtCore.Qt.AlignCenter)
        self.label_3.setObjectName("label_3")
        self.gridLayout_2.addWidget(self.label_3, 3, 2, 1, 1)
        self.label_2 = QtWidgets.QLabel(self.inputDataFrame)
        self.label_2.setStyleSheet("border: none;")
        self.label_2.setAlignment(QtCore.Qt.AlignCenter)
        self.label_2.setObjectName("label_2")
        self.gridLayout_2.addWidget(self.label_2, 3, 6, 1, 1)
        self.comboPrefixCycle = QtWidgets.QComboBox(self.inputDataFrame)
        self.comboPrefixCycle.setObjectName("comboPrefixCycle")
        self.comboPrefixCycle.addItem("")
        self.comboPrefixCycle.addItem("")
        self.gridLayout_2.addWidget(self.comboPrefixCycle, 4, 6, 1, 1)
        self.comboMCS = QtWidgets.QComboBox(self.inputDataFrame)
        self.comboMCS.setObjectName("comboMCS")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.comboMCS.addItem("")
        self.gridLayout_2.addWidget(self.comboMCS, 4, 2, 1, 1)
        self.label_4 = QtWidgets.QLabel(self.inputDataFrame)
        self.label_4.setStyleSheet("border: none;")
        self.label_4.setAlignment(QtCore.Qt.AlignCenter)
        self.label_4.setObjectName("label_4")
        self.gridLayout_2.addWidget(self.label_4, 3, 1, 1, 1)
        self.ComboMimo = QtWidgets.QComboBox(self.inputDataFrame)
        self.ComboMimo.setObjectName("ComboMimo")
        self.ComboMimo.addItem("")
        self.ComboMimo.addItem("")
        self.ComboMimo.addItem("")
        self.ComboMimo.addItem("")
        self.gridLayout_2.addWidget(self.ComboMimo, 4, 4, 1, 1)
        self.label_6 = QtWidgets.QLabel(self.inputDataFrame)
        self.label_6.setStyleSheet("border: none;")
        self.label_6.setAlignment(QtCore.Qt.AlignCenter)
        self.label_6.setObjectName("label_6")
        self.gridLayout_2.addWidget(self.label_6, 3, 8, 1, 1)
        self.line_2 = QtWidgets.QFrame(self.inputDataFrame)
        self.line_2.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_2.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_2.setObjectName("line_2")
        self.gridLayout_2.addWidget(self.line_2, 1, 1, 1, 1)
        self.comboCarrierAgg = QtWidgets.QComboBox(self.inputDataFrame)
        self.comboCarrierAgg.setEnabled(True)
        self.comboCarrierAgg.setObjectName("comboCarrierAgg")
        self.comboCarrierAgg.addItem("")
        self.comboCarrierAgg.addItem("")
        self.comboCarrierAgg.addItem("")
        self.comboCarrierAgg.addItem("")
        self.comboCarrierAgg.addItem("")
        self.gridLayout_2.addWidget(self.comboCarrierAgg, 4, 8, 1, 1)
        self.label_9 = QtWidgets.QLabel(self.inputDataFrame)
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_9.setFont(font)
        self.label_9.setStyleSheet("border: none;")
        self.label_9.setObjectName("label_9")
        self.gridLayout_2.addWidget(self.label_9, 0, 1, 1, 1)
        self.gridLayout.addWidget(self.inputDataFrame, 2, 0, 2, 2)
        self.label_10 = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI Black")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_10.setFont(font)
        self.label_10.setAlignment(QtCore.Qt.AlignCenter)
        self.label_10.setObjectName("label_10")
        self.gridLayout.addWidget(self.label_10, 6, 0, 1, 2)
        self.outputParamsDataFrame = QtWidgets.QFrame(self.centralwidget)
        self.outputParamsDataFrame.setEnabled(True)
        self.outputParamsDataFrame.setStyleSheet("border: 2px solid #8f8f91;")
        self.outputParamsDataFrame.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.outputParamsDataFrame.setFrameShadow(QtWidgets.QFrame.Raised)
        self.outputParamsDataFrame.setObjectName("outputParamsDataFrame")
        self.gridLayout_3 = QtWidgets.QGridLayout(self.outputParamsDataFrame)
        self.gridLayout_3.setObjectName("gridLayout_3")
        self.outputTbsIndex = QtWidgets.QLineEdit(self.outputParamsDataFrame)
        self.outputTbsIndex.setEnabled(False)
        self.outputTbsIndex.setObjectName("outputTbsIndex")
        self.gridLayout_3.addWidget(self.outputTbsIndex, 4, 1, 1, 1)
        self.ouputTbsValue = QtWidgets.QLineEdit(self.outputParamsDataFrame)
        self.ouputTbsValue.setEnabled(False)
        self.ouputTbsValue.setObjectName("ouputTbsValue")
        self.gridLayout_3.addWidget(self.ouputTbsValue, 4, 2, 1, 1)
        self.label_13 = QtWidgets.QLabel(self.outputParamsDataFrame)
        self.label_13.setStyleSheet("border: none;")
        self.label_13.setAlignment(QtCore.Qt.AlignCenter)
        self.label_13.setObjectName("label_13")
        self.gridLayout_3.addWidget(self.label_13, 2, 4, 1, 2)
        self.outputPRB = QtWidgets.QLineEdit(self.outputParamsDataFrame)
        self.outputPRB.setEnabled(False)
        self.outputPRB.setObjectName("outputPRB")
        self.gridLayout_3.addWidget(self.outputPRB, 4, 0, 1, 1)
        self.label_17 = QtWidgets.QLabel(self.outputParamsDataFrame)
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_17.setFont(font)
        self.label_17.setStyleSheet("border: none;")
        self.label_17.setObjectName("label_17")
        self.gridLayout_3.addWidget(self.label_17, 0, 0, 1, 1)
        self.outputNumRE = QtWidgets.QLineEdit(self.outputParamsDataFrame)
        self.outputNumRE.setEnabled(False)
        self.outputNumRE.setObjectName("outputNumRE")
        self.gridLayout_3.addWidget(self.outputNumRE, 4, 6, 1, 1)
        self.label_11 = QtWidgets.QLabel(self.outputParamsDataFrame)
        self.label_11.setStyleSheet("border: none;")
        self.label_11.setAlignment(QtCore.Qt.AlignCenter)
        self.label_11.setObjectName("label_11")
        self.gridLayout_3.addWidget(self.label_11, 2, 0, 1, 1)
        self.label_12 = QtWidgets.QLabel(self.outputParamsDataFrame)
        self.label_12.setStyleSheet("border: none;")
        self.label_12.setAlignment(QtCore.Qt.AlignCenter)
        self.label_12.setObjectName("label_12")
        self.gridLayout_3.addWidget(self.label_12, 2, 2, 1, 2)
        self.label_15 = QtWidgets.QLabel(self.outputParamsDataFrame)
        self.label_15.setStyleSheet("border: none;")
        self.label_15.setAlignment(QtCore.Qt.AlignCenter)
        self.label_15.setObjectName("label_15")
        self.gridLayout_3.addWidget(self.label_15, 2, 6, 1, 2)
        self.label_14 = QtWidgets.QLabel(self.outputParamsDataFrame)
        self.label_14.setStyleSheet("border: none;")
        self.label_14.setAlignment(QtCore.Qt.AlignCenter)
        self.label_14.setObjectName("label_14")
        self.gridLayout_3.addWidget(self.label_14, 2, 1, 1, 1)
        self.outputModulacao = QtWidgets.QLineEdit(self.outputParamsDataFrame)
        self.outputModulacao.setEnabled(False)
        self.outputModulacao.setObjectName("outputModulacao")
        self.gridLayout_3.addWidget(self.outputModulacao, 4, 4, 1, 1)
        self.label_16 = QtWidgets.QLabel(self.outputParamsDataFrame)
        self.label_16.setStyleSheet("border: none;")
        self.label_16.setAlignment(QtCore.Qt.AlignCenter)
        self.label_16.setObjectName("label_16")
        self.gridLayout_3.addWidget(self.label_16, 2, 8, 1, 1)
        self.outputQtdSimbolos = QtWidgets.QLineEdit(self.outputParamsDataFrame)
        self.outputQtdSimbolos.setEnabled(False)
        self.outputQtdSimbolos.setObjectName("outputQtdSimbolos")
        self.gridLayout_3.addWidget(self.outputQtdSimbolos, 4, 8, 1, 1)
        self.line_3 = QtWidgets.QFrame(self.outputParamsDataFrame)
        self.line_3.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_3.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_3.setObjectName("line_3")
        self.gridLayout_3.addWidget(self.line_3, 1, 0, 1, 1)
        self.gridLayout.addWidget(self.outputParamsDataFrame, 5, 0, 1, 2)
        self.btnCalcTPut = QtWidgets.QPushButton(self.centralwidget)
        self.btnCalcTPut.setEnabled(True)
        self.btnCalcTPut.setStyleSheet("QPushButton {\n"
"      border: 2px solid #4CAF50;\n"
"      border-radius: 6px;\n"
"      background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,\n"
"                                        stop: 0 #f6f7fa, stop: 1 #dadbde);\n"
"      min-width: 80px;\n"
"      padding: 15px 32px;\n"
"  }\n"
"\n"
"QPushButton:pressed {\n"
"      background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,\n"
"                                        stop: 0 #dadbde, stop: 1 #f6f7fa);\n"
"  }")
        self.btnCalcTPut.setObjectName("btnCalcTPut")
        self.gridLayout.addWidget(self.btnCalcTPut, 8, 0, 1, 2)
        self.label = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI Black")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setAlignment(QtCore.Qt.AlignCenter)
        self.label.setObjectName("label")
        self.gridLayout.addWidget(self.label, 0, 0, 1, 2)
        self.outputTabela = QtWidgets.QLineEdit(self.centralwidget)
        self.outputTabela.setEnabled(False)
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.outputTabela.setFont(font)
        self.outputTabela.setStyleSheet("border: 2px solid #8f8f91; padding: 2px 2px;")
        self.outputTabela.setAlignment(QtCore.Qt.AlignCenter)
        self.outputTabela.setObjectName("outputTabela")
        self.gridLayout.addWidget(self.outputTabela, 11, 0, 1, 2)
        self.outputEquacao = QtWidgets.QLineEdit(self.centralwidget)
        self.outputEquacao.setEnabled(False)
        font = QtGui.QFont()
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.outputEquacao.setFont(font)
        self.outputEquacao.setStyleSheet("border: 2px solid #8f8f91; padding: 2px 2px;")
        self.outputEquacao.setText("")
        self.outputEquacao.setAlignment(QtCore.Qt.AlignCenter)
        self.outputEquacao.setObjectName("outputEquacao")
        self.gridLayout.addWidget(self.outputEquacao, 16, 0, 1, 2)
        self.label_7 = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_7.setFont(font)
        self.label_7.setStyleSheet("")
        self.label_7.setAlignment(QtCore.Qt.AlignCenter)
        self.label_7.setObjectName("label_7")
        self.gridLayout.addWidget(self.label_7, 9, 0, 1, 2)
        self.label_8 = QtWidgets.QLabel(self.centralwidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.label_8.setFont(font)
        self.label_8.setAlignment(QtCore.Qt.AlignCenter)
        self.label_8.setObjectName("label_8")
        self.gridLayout.addWidget(self.label_8, 12, 0, 1, 2)
        self.line = QtWidgets.QFrame(self.centralwidget)
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.gridLayout.addWidget(self.line, 17, 0, 1, 2)
        MainWindow.setCentralWidget(self.centralwidget)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.label_5.setText(_translate("MainWindow", "Mimo"))
        self.comboBanda.setItemText(0, _translate("MainWindow", "1.4"))
        self.comboBanda.setItemText(1, _translate("MainWindow", "3"))
        self.comboBanda.setItemText(2, _translate("MainWindow", "5"))
        self.comboBanda.setItemText(3, _translate("MainWindow", "10"))
        self.comboBanda.setItemText(4, _translate("MainWindow", "15"))
        self.comboBanda.setItemText(5, _translate("MainWindow", "20"))
        self.label_3.setText(_translate("MainWindow", "MCS"))
        self.label_2.setText(_translate("MainWindow", "CP"))
        self.comboPrefixCycle.setItemText(0, _translate("MainWindow", "Normal"))
        self.comboPrefixCycle.setItemText(1, _translate("MainWindow", "Extendido"))
        self.comboMCS.setItemText(0, _translate("MainWindow", "0"))
        self.comboMCS.setItemText(1, _translate("MainWindow", "1"))
        self.comboMCS.setItemText(2, _translate("MainWindow", "2"))
        self.comboMCS.setItemText(3, _translate("MainWindow", "3"))
        self.comboMCS.setItemText(4, _translate("MainWindow", "4"))
        self.comboMCS.setItemText(5, _translate("MainWindow", "5"))
        self.comboMCS.setItemText(6, _translate("MainWindow", "6"))
        self.comboMCS.setItemText(7, _translate("MainWindow", "7"))
        self.comboMCS.setItemText(8, _translate("MainWindow", "8"))
        self.comboMCS.setItemText(9, _translate("MainWindow", "9"))
        self.comboMCS.setItemText(10, _translate("MainWindow", "10"))
        self.comboMCS.setItemText(11, _translate("MainWindow", "11"))
        self.comboMCS.setItemText(12, _translate("MainWindow", "12"))
        self.comboMCS.setItemText(13, _translate("MainWindow", "13"))
        self.comboMCS.setItemText(14, _translate("MainWindow", "14"))
        self.comboMCS.setItemText(15, _translate("MainWindow", "15"))
        self.comboMCS.setItemText(16, _translate("MainWindow", "16"))
        self.comboMCS.setItemText(17, _translate("MainWindow", "17"))
        self.comboMCS.setItemText(18, _translate("MainWindow", "18"))
        self.comboMCS.setItemText(19, _translate("MainWindow", "19"))
        self.comboMCS.setItemText(20, _translate("MainWindow", "20"))
        self.comboMCS.setItemText(21, _translate("MainWindow", "21"))
        self.comboMCS.setItemText(22, _translate("MainWindow", "22"))
        self.comboMCS.setItemText(23, _translate("MainWindow", "23"))
        self.comboMCS.setItemText(24, _translate("MainWindow", "24"))
        self.comboMCS.setItemText(25, _translate("MainWindow", "25"))
        self.comboMCS.setItemText(26, _translate("MainWindow", "26"))
        self.comboMCS.setItemText(27, _translate("MainWindow", "27"))
        self.comboMCS.setItemText(28, _translate("MainWindow", "28"))
        self.label_4.setText(_translate("MainWindow", "Banda (MHz)"))
        self.ComboMimo.setItemText(0, _translate("MainWindow", "1"))
        self.ComboMimo.setItemText(1, _translate("MainWindow", "2"))
        self.ComboMimo.setItemText(2, _translate("MainWindow", "4"))
        self.ComboMimo.setItemText(3, _translate("MainWindow", "8"))
        self.label_6.setText(_translate("MainWindow", "CA"))
        self.comboCarrierAgg.setItemText(0, _translate("MainWindow", "1"))
        self.comboCarrierAgg.setItemText(1, _translate("MainWindow", "2"))
        self.comboCarrierAgg.setItemText(2, _translate("MainWindow", "3"))
        self.comboCarrierAgg.setItemText(3, _translate("MainWindow", "4"))
        self.comboCarrierAgg.setItemText(4, _translate("MainWindow", "5"))
        self.label_9.setText(_translate("MainWindow", "Parametros de Entrada"))
        self.label_10.setText(_translate("MainWindow", "Taxa de Transmisao LTE Release 10"))
        self.label_13.setText(_translate("MainWindow", "Modulacao"))
        self.label_17.setText(_translate("MainWindow", "Parametros de Saida"))
        self.label_11.setText(_translate("MainWindow", "PRB"))
        self.label_12.setText(_translate("MainWindow", "Valor TBS"))
        self.label_15.setText(_translate("MainWindow", "N RE"))
        self.label_14.setText(_translate("MainWindow", "TBS Index"))
        self.label_16.setText(_translate("MainWindow", "Qtd. Simbolos"))
        self.btnCalcTPut.setText(_translate("MainWindow", "Calcular"))
        self.label.setText(_translate("MainWindow", "Calculadora LTE"))
        self.label_7.setText(_translate("MainWindow", "Tabela"))
        self.label_8.setText(_translate("MainWindow", "Equacao"))
